skimage/future/graph/tests/test_rag.py

```diff
@@ -184,6 +187,36 @@ def test_ncut_stable_subgraph():
     assert new_labels.max() == 0
 
 
+def test_inplace():
+    """Make sure normalized cuts does not modify arguments
+    when in_place=False"""
+    img = data.coffee()
+    labels1 = segmentation.slic(img, compactness=30, n_segments=400)
+    g = graph.rag_mean_color(img, labels1, mode='similarity')
+    backup_labels = deepcopy(labels1)
+    backup_g = deepcopy(g)
+    _ = graph.cut_normalized(labels1, g, in_place=False, thresh=1e-8)
+    assert_array_equal(backup_labels, labels1)
+    assert backup_g == g
+
+
+def test_random_seed():
+    """ensure cut_normalized returns the same output for the same input,
+    when setting numpy random seed
+    """
+    img = data.coffee()
+    labels1 = segmentation.slic(img, compactness=30, n_segments=400)
+    g = graph.rag_mean_color(img, labels1, mode='similarity')
+    results = [None] * 10
+    for i in range(len(results)):
+        np.random.seed(1234)
```

Comment:
this test failed when I used `random.seed` instead

Reply:
apparently this is to be expected: https://stackoverflow.com/questions/31057197/should-i-use-random-seed-or-numpy-random-seed-to-control-random-number-gener